src/test/ui/binding/issue-53114-borrow-checks.rs

```diff
@@ -0,0 +1,58 @@
+// Issue #53114: NLL's borrow check had some deviations from the old borrow
+// checker, and both had some deviations from our ideal state. This test
+// captures the behavior of how `_` bindings are handled wiith respect to how we
+// flag expressions that are meant to request unsafe blocks.
+
+struct M;
+
+fn let_wild_gets_moved_expr() {
+    let m = M;
+    drop(m);
+    let _ = m; // accepted, and want it to continue to be
+
+    let mm = (M, M); // variation on above with `_` in substructure
+    let (_x, _) = mm;
+    let (_, _y) = mm;
+    let (_, _) = mm;
+}
+
+fn match_moved_expr_to_wild() {
```

Comment:
General note: Would be good to also test `if let` as it might diverge somewhat in the future in terms of MIR building.

Reply:
hmm. interesting the most obvious "ports" of these tests as written into `if let` form runs afoul of "warning: irrefutable if-let pattern"

Which makes total sense.

But also leads me to wonder whether I should be testing them in different fashion. I'm not sure these tests translate *over* to refutable patterns, because if the data has been moved away, then you won't have a discriminant to inspect.

Reply:
Perhaps just inserting an `Option<_>` layer would be good for the match/if-let tests? iirc I did that in a somewhat similar test somewhere. Presumably `_` should work below sum types as well.